src/jestUtils.ts

```diff
@@ -0,0 +1,359 @@
+  [P in keyof T]: T[P] & { numberOfPointers: number };
+};
+type DefaultEventsMapping = WithNumberOfPointers<{
+  [flingHandlerName]: FlingGestureHandlerEventPayload;
+  [forceTouchHandlerName]: ForceTouchGestureHandlerEventPayload;
+  [longPressHandlerName]: LongPressGestureHandlerEventPayload;
+  [nativeViewHandlerName]: NativeViewGestureHandlerPayload;
+  [panHandlerName]: PanGestureHandlerEventPayload;
+  [pinchHandlerName]: PinchGestureHandlerEventPayload;
+  [rotationHandlerName]: RotationGestureHandlerEventPayload;
+  [tapHandlerName]: TapGestureHandlerEventPayload;
+}>;
+
+const handlersDefaultEvents: DefaultEventsMapping = {
+  [flingHandlerName]: {
+    x: 0,
+    y: 0,
+    absoluteX: 0,
+    absoluteY: 0,
+    numberOfPointers: 1,
+  },
+  [forceTouchHandlerName]: {
+    x: 0,
+    y: 0,
+    absoluteX: 0,
+    absoluteY: 0,
+    force: 1,
+    numberOfPointers: 1,
+  },
+  [longPressHandlerName]: {
+    x: 0,
+    y: 0,
+    absoluteX: 0,
+    absoluteY: 0,
+    duration: 100,
+    numberOfPointers: 1,
+  },
+  [nativeViewHandlerName]: {
+    pointerInside: true,
+    numberOfPointers: 1,
+  },
+  [panHandlerName]: {
+    x: 0,
+    y: 0,
+    absoluteX: 0,
+    absoluteY: 0,
+    translationX: 100,
+    translationY: 0,
+    velocityX: 3,
+    velocityY: 0,
+    numberOfPointers: 1,
+  },
+  [pinchHandlerName]: {
+    focalX: 0,
+    focalY: 0,
+    scale: 2,
+    velocity: 1,
+    numberOfPointers: 2,
+  },
+  [rotationHandlerName]: {
+    anchorX: 0,
+    anchorY: 0,
+    rotation: 3.14,
+    velocity: 2,
+    numberOfPointers: 2,
+  },
+  [tapHandlerName]: {
+    x: 0,
+    y: 0,
+    absoluteX: 0,
+    absoluteY: 0,
+    numberOfPointers: 1,
+  },
+};
+
+function isGesture(
+  componentOrGesture: ReactTestInstance | GestureType
+): componentOrGesture is GestureType {
+  return componentOrGesture instanceof BaseGesture;
+}
+
+function wrapWithNativeEvent(event: GestureHandlerTestEvent) {
+  return { nativeEvent: event };
+}
+
+function fillOldStateChanges(
+  previousEvent: GestureHandlerTestEvent | null,
+  currentEvent: Omit<GestureHandlerTestEvent, 'oldState'>
+): GestureHandlerTestEvent {
+  const isFirstEvent = previousEvent === null;
+  if (isFirstEvent) {
+    return {
+      oldState: State.UNDETERMINED,
+      ...currentEvent,
+    } as GestureHandlerTestEvent;
+  }
+
+  const isGestureStateEvent = previousEvent.state !== currentEvent.state;
+  if (isGestureStateEvent) {
+    return {
+      oldState: previousEvent.state,
+      ...currentEvent,
+    } as GestureHandlerTestEvent;
+  } else {
+    return currentEvent as GestureHandlerTestEvent;
+  }
+}
+
+function fillMissingActiveStateFields(
+  previousEvent: Omit<GestureHandlerTestEvent, 'oldState'> | null,
+  currentEvent: Omit<GestureHandlerTestEvent, 'state' | 'oldState'>
+) {
+  const isFirstEvent = previousEvent === null;
+  if (isFirstEvent) {
+    return currentEvent;
+  }
+  if (previousEvent.state === State.ACTIVE && !currentEvent['state']) {
```

Comment:
Same here, `previousEvent` may be null.
```suggestion
  if (previousEvent?.state === State.ACTIVE && !currentEvent['state']) {
```

Reply:
Can it? I think we check for null above, right?
```tsx
const isFirstEvent = previousEvent === null;
if (isFirstEvent) {
  return currentEvent;
}
```

Reply:
But `!currentEvent['state']` is buggy, it's true for `state === 0`. I'll change it.

Reply:
Oh, you're right. I've missed it.

Reply:
Property check fixed in 4d3dd6d5.